### PR TITLE
feat: [sc-154163] update Android SDK hello app readme and comments to standard design

### PR DESCRIPTION
**Pull request by ember-stevens**
After receiving some feedback that it'd be helpful if the Hello apps were more standardized, Docs team is working on updating them to [match the spec](https://launchdarkly.atlassian.net/wiki/spaces/ENG/pages/1499562073/SDK+hello+world).

For this app, some readme updates and adding standard comments. `MainActivity.java` is missing initialization success and failure messages.

Story details: https://app.shortcut.com/launchdarkly/story/154163

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+2. Edit `MainActivity.java`
 in the `src/main/java/com/launchdarkly/hello_android` directory and set the value of MOBILE_KEY to your LaunchDarkly mobile key. If there is an existing boolean feature flag in your LaunchDarkly project that you want to evaluate, set FLAG_KEY to the flag key.
+
+```
+.mobileKey("123456abcdef")
+
+private static final String FLAG_KEY = "my-boolean-flag";
+```
+
+3. Run your application through the Emulator or on a real device.
```
Comment (ember-stevens) [inline README.md:16]: This doesn't match the spec, but not sure of the command to run here.

File: app/src/main/java/com/launchdarkly/hello_android/MainActivity.java [hunk 2]
```diff
@@ -9,29 +9,32 @@
...
+// Set up the user properties. This user should appear on your LaunchDarkly users dashboard soon after you run the demo.
+        LDUser user = new LDUser.Builder("example-user-key")
+                .name("Sandy")
                 .build();
 
         LDClient client = LDClient.init(getApplication(), ldConfig, user, 5);
 
         if (client.boolVariation(FLAG_KEY, false)) {
-            Log.i(TAG, "Showing feature");
+            Log.i(TAG, "Feature flag FLAG_KEY is true for this user");
```
Comment (ember-stevens) [inline app/src/main/java/com/launchdarkly/hello_android/MainActivity.java]: Will need assistance with message syntax.

File: app/src/main/java/com/launchdarkly/hello_android/MainActivity.java [hunk 3]
```diff
@@ -9,29 +9,30 @@
...
+                .name("Sandy")
                 .build();
 
         LDClient client = LDClient.init(getApplication(), ldConfig, user, 5);
 
         if (client.boolVariation(FLAG_KEY, false)) {
-            Log.i(TAG, "Showing feature");
-        } else {
-            Log.i(TAG, "Not Showing feature");
+            Log.i(TAG, "Feature flag '" + FLAG_KEY + "' is " + flagValue + " for this user");
```
Comment (ember-stevens) [inline app/src/main/java/com/launchdarkly/hello_android/MainActivity.java]: I took this formatting from hello-java.

File: README.md (hunk 1 above)
Comment (mmrj) [reply README.md:16]: I think this is fine. We have something similar in https://github.com/launchdarkly/hello-roku, for example.

File: app/src/main/java/com/launchdarkly/hello_android/MainActivity.java [hunk 4]
```diff
@@ -9,30 +9,30 @@
...
 
         LDClient client = LDClient.init(getApplication(), ldConfig, user, 5);
 
-        if (client.boolVariation(FLAG_KEY, false)) {
-            Log.i(TAG, "Showing feature");
-        } else {
-            Log.i(TAG, "Not Showing feature");
-        }
+        boolean flagValue = client.boolVariation(FLAG_KEY, false)
+                Log.i(TAG, "Feature flag '" + FLAG_KEY + "' is " + flagValue + " for this user");
```
Comment (mmrj) [inline app/src/main/java/com/launchdarkly/hello_android/MainActivity.java]: Saw your earlier comment...I'm not sure if this syntax is correct either. It's possibly 
```
Log.i(TAG, "Feature flag [%s] is [%s] for this user", FLAG_KEY, flagValue)
```
based on a quick search in the [sdk repo](https://github.com/launchdarkly/android-client-sdk/search?q=log). Not sure though.

**Review by mmrj — COMMENTED** [review]
LGTM! Will wait to approve until the SDK team can confirm syntax for the log statement.

**Review by louis-launchdarkly — APPROVED** [review]

**Pull request merged**